lib/semantics/resolve-names.cc

```diff
@@ -1547,8 +1558,8 @@ bool ModuleVisitor::Pre(const parser::Module &x) {
   const auto &name{
       std::get<parser::Statement<parser::ModuleStmt>>(x.t).statement.v.source};
   auto &subpPart{std::get<std::optional<parser::ModuleSubprogramPart>>(x.t)};
-  auto &symbol{BeginModule(name, false, subpPart)};
-  MakeSymbol(name, symbol.details());
+  BeginModule(name, false, subpPart);
```

Comment:
Should BeginModule() still be returning a Symbol& ?

Reply:
We're still using the return value in the submodule case, but it's possible it could follow the pattern of the module case.

Reply:
Regarding symbol ownership: Yes, Symbols are all owned by Scopes, which are in a tree rooted at the global scope, which is in the SemanticsContext. That's why we can freely use pointers and references to Symbols and Scopes without worrying about lifetimes.